samples/snippets/visualbasic/VS_Snippets_Wpf/UIAFragmentProvider_snip/VisualBasic/ListFragment.vb

```diff
@@ -209,7 +209,7 @@ Class ParentList
     Protected Overrides Sub WndProc(ByRef m As Message)
         Const WM_GETOBJECT As Integer = &H3D
 
-        If m.Msg = WM_GETOBJECT AndAlso m.LParam.ToInt32() = AutomationInteropProvider.RootObjectId Then
+        If m.Msg = WM_GETOBJECT AndAlso CType(CType(m.LParam, Long), Int32) = AutomationInteropProvider.RootObjectId Then
```

Comment:
@KlausLoeffelmann may I get you to confirm this?

Reply:
Ping @KlausLoeffelmann

Reply:
Sorry for the late reply.
Why does `ToInt32` not work here? Wrong conversion of the sign-bit?
Anyway, if there is a need to first convert into `Long` and then `Int`, I'd do it with `CInt` and `CLng`, so, from Memory it would be:

```suggestion
        If m.Msg = WM_GETOBJECT AndAlso CInt(CLng(m.LParam)) = AutomationInteropProvider.RootObjectId Then
```

Would that work?

Reply:
> Why does `ToInt32` not work here? Wrong conversion of the sign-bit?

Please refer to https://source.dot.net/#System.Private.CoreLib/IntPtr.cs,102.